changelog.d/10316.misc

```diff
@@ -0,0 +1 @@
+Rebuild event context and auth when processing specific results from `ThirdPartyEventRules` modules.
```

Comment:
> when processing specific results

This is a little vague. How about "Rebuild context for events that have been modified by `ThirdPartEventRules` modules"?

Reply:
That doesn't work either, because modules can just return the event dict without modifying it to trick Synapse into rebuilding the context and auth (which is what we do in https://github.com/matrix-org/synapse-dinsic/pull/100)